clang_bind/parse.py

```diff
@@ -1,134 +1,68 @@
-                **cursor_kind_utils.get_properties_dict(),  # Properties
-            },
-            "cursor": {
-                **cursor_utils.get_check_functions_dict(),
-                **cursor_utils.get_get_functions_dict(),
-                **cursor_utils.get_properties_dict(),
-            },
-            "type": {
-                **cursor_type_utils.get_check_functions_dict(),
-                **cursor_type_utils.get_get_functions_dict(),
-                **cursor_type_utils.get_properties_dict(),
-            },
-            "members": [],
-        }
-
-        # HACKY FIXES
-        # get spelling from object
-        parsed_node["cursor"]["result_type"] = parsed_node["cursor"][
-            "result_type"
-        ].spelling
-        # replace `AccessSpecifier.value` with just `value`
-        parsed_node["cursor"]["access_specifier"] = parsed_node["cursor"][
-            "access_specifier"
-        ].name
-        # replace `TypeKind.value` with just `value`
-        parsed_node["type"]["kind"] = parsed_node["type"]["kind"].name
-
-        return parsed_node
-
-    @classmethod
-    def parse_node_recursive(cls, node):
+    def _is_valid_child(self, clang_child_cursor):
         """
-        Generates parsed information by recursively traversing the AST
-
-        Parameters:
-            - node (dict):
-                - The node in the AST
-                - Keys:
-                    - cursor: The cursor pointing to a node
-                    - filename:
-                        - The file's name to check if the node belongs to it
-                        - Needed to ensure that only symbols belonging to the file gets parsed, not the included files' symbols
-                    - depth: The depth of the node (root=0)
-
-        Returns:
-            - parsed_info (dict):
-                - Contains key-value pairs of various traits of a node
-                - The key 'members' contains the node's children's `parsed_info`
+        Check if the child is valid (child should be in the same file as the parent).
         """
+        return (
+            clang_child_cursor.location.file
+            and clang_child_cursor.location.file.name == self.filename
+        )
 
-        cursor = node.get("cursor")
-        filename = node.get("filename")
-        depth = node.get("depth")
-
-        parsed_info = cls.get_parsed_node(node)
-
-        # Get cursor's children and recursively add their info to a dictionary, as members of the parent
-        for child in cursor.get_children():
-            child_node = {"cursor": child, "filename": filename, "depth": depth + 1}
-            if cls._is_valid_child(node, child_node):
-                child_parsed_info = cls.parse_node_recursive(child_node)
-                parsed_info["members"].append(child_parsed_info)
-
-        return parsed_info
-
-    def get_parsed_info(self):
+    def _construct_tree(self, node):
```

Comment:
Same here? Make construct tree a free function and get_tree provides an additional argument.

Reason: functionality is independent of class. Extracting out allows a smaller to parse class

Reply:
Even if we pass the arguments, we need to call another method either via `self` or `cls`:
https://github.com/PointCloudLibrary/clang-bind/blob/df6cd5809cae7d66b0f2e7538aba6fcc0bddd81d/clang_bind/parse.py#L60